src/Compiler/Checking/import.fs

```diff
@@ -90,6 +93,19 @@ type [<Struct; NoComparison; CustomEquality>] TTypeCacheKey =
 
         combined
 
+//let typeSubsumptionCaches = ConditionalWeakTable<_, Cache<TTypeCacheKey, bool>>()
+
+let typeSubsumptionCache =
+    //typeSubsumptionCaches.GetValue(g, fun _ ->
+        Cache<TTypeCacheKey, bool>.Create(
+            { CacheOptions.Default with
```

Comment:
Could the compilationMode flow into here, and in case of oneoff, just not do the background tasks at all?

But I understand that ideally we would like to get rid off lastAccessed and accessCount maintenance as well, so a dispatcher on higher level to plain ConcurrentDictionary would be prefered.

Reply:
Thinking out loud - the compilation mode is a property of the process (fsc.exe vs. being part of devenv.exe or similar).

Once JITed, it will never vary - I think we could rely on de-virtualization done at JIT time here, equivalently any branching based on a global boolean flag.

Reply:
This shouldn't be global static, but maybe a prop of TcImports, with lazy initialization.  We probably have compilationMode available there and can provide appropriate implementation as required.

Reply:
> Could the compilationMode flow into here, and in case of oneoff, just not do the background tasks at all?
> 
> 
> 
> But I understand that ideally we would like to get rid off lastAccessed and accessCount maintenance as well, so a dispatcher on higher level to plain ConcurrentDictionary would be prefered.

Yeah, no need for background task in the one-off at all. I would even argue that eviction is needed at all in this case. Currently it's just forever growing CD, so in case of cache, you can just make its size a max int (and static for compilation, since we rely on stamps anyway).